tools/GraphClust/CollectResults/evaluation.py

```diff
@@ -10,56 +10,65 @@ def sh(script):
 dataNames = "FASTA/data.names"
 
 listOfClusters = []
-listOfClasses = []
+listOfHeaders = []
+headersNames = set()
 cluster_seqs_stats_path = "RESULTS/*.cluster.all"
 cluster_seqs_stats_files = glob.glob(cluster_seqs_stats_path)
 
+with open(dataNames, "r") as names:
+    for line2 in names.readlines():
+        splits2 = line2.split()
+        fullHeader = ''
+        if len(splits2) >= 6:
+            fullHeader = splits2[5]
+            headersNames.add(fullHeader)
+
 blackList = []
 numberOfClusters = 0
 for singleFile in sorted(cluster_seqs_stats_files):
     numberOfClusters += 1
     with open(singleFile, "r") as f:
         for line in f.readlines():
-            uniqueId = line.split()[8]
-            clustNum = line.split()[2]
-            rnaClass, sep, tail = uniqueId.partition("_")
-            listOfClasses.append(rnaClass)
+            splits = line.split()
+            header = ''
+            if len(splits) >= 11:
+                header = splits[10]
+            clustNum = splits[2]
+            listOfHeaders.append(header)
             listOfClusters.append(clustNum)
-            with open(dataNames, "r") as names:
-                for line in names.readlines():
-                    fullUniqeId = line.split()[3]
-                    rnaClass, sep, tail = fullUniqeId.partition("_")
-                    if fullUniqeId == uniqueId:
-                        blackList.append(uniqueId)
+            if header in headersNames:
+                blackList.append(header)
 
 numberOfClusters += 1  # 1 cluster for all unassigned seqs
 with open(dataNames, "r") as names:
     for line in names.readlines():
         fullUniqeId = line.split()[3]
-        rnaClass, sep, tail = fullUniqeId.partition("_")
-        rnaClass, sep, tail = fullUniqeId.partition("_")
-        if fullUniqeId not in blackList:
-            listOfClasses.append(rnaClass)
+        fullHeader = ''
+        if len(line.split()) >= 6:
+            fullHeader = line.split()[5]
```

Comment:
if you move the split of of the condition and save it, you don't need to do it twice